sync/pgxn.js

```diff
@@ -0,0 +1,140 @@
+(async function() {
+  'use strict';
+
+const { URL } = require('url');
+const fs = require('fs');
+
+if (process.argv.some(a => (a == '-h' || a == '--help'))) {
+  console.log(`Exctracts PL/pgSQL extensions for Sourcerer awesome-libraries from pgnx.org - PostgreSQL extension network.`);
+  console.log(`Usage: node pgnx.js [--debug]`);
+  return;
+}
+
+let debugMode = process.argv.some(a => a =='--debug');
+
+const libFile = `${__dirname}/../libs/plpgsql.json`;
+let libContent = fs.readFileSync(libFile, 'utf-8');
+
+let libs = null;
+try {
+  libs = JSON.parse(libContent);
+}
+catch(e) {
+  throw new Error(`${libFile} is not valid JSON; ${e.toString()}`);
+}
+libs = new Set(libs.map(l => l.name));
+
+let request = (url) => {
```

Comment:
Surely there is a shorter way of loading a resource by its url in Javascript?

Reply:
not I am aware of, afaik there's no handy way to load url in native node. There are 3d party packages like 'request', but it require installation (including package manager). I moved the function into util.js to keep it off the main file.